src/bin/mux/mod.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2017-2018, The rav1e contributors. All rights reserved
+//
+// This source code is subject to the terms of the BSD 2 Clause License and
+// the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+// was not distributed with this source code in the LICENSE file, you can
+// obtain it at www.aomedia.org/license/software. If the Alliance for Open
+// Media Patent License 1.0 was not distributed with this source code in the
+// PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+
+mod muxer;
+pub use muxer::*;
+
+mod ivfmuxer;
+use ivfmuxer::IvfMuxer;
+
+mod y4mmuxer;
+pub use y4mmuxer::write_y4m_frame;
+
+#[cfg(feature = "ffmpeg-sys")]
+mod avformatmuxer;
+#[cfg(feature = "ffmpeg-sys")]
+use avformatmuxer::AvformatMuxer;
+
+use std::ffi::OsStr;
+use std::path::Path;
+
+fn need_container(path: &str) -> bool {
+  let ext =
```

Comment:
This function is still wrong. You do not want to produce an `ivf`  if the extension is not supported.

Reply:
ok, I will change to this